sys/net/application_layer/gcoap/forward_proxy.c

```diff
@@ -0,0 +1,357 @@
+/*
+ * Copyright (C) 2020 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @author  Cenk Gündoğan <cenk.guendogan@haw-hamburg.de>
+ */
+
+#include "net/gcoap.h"
+#include "net/gcoap/forward_proxy.h"
+#include "uri_parser.h"
+
+#define ENABLE_DEBUG    0
+#include "debug.h"
+
+typedef struct {
+    int in_use;
+    sock_udp_ep_t ep;
+} client_ep_t;
+
+static uint8_t proxy_req_buf[CONFIG_GCOAP_PDU_BUF_SIZE];
+static client_ep_t _client_eps[CONFIG_GCOAP_REQ_WAITING_MAX];
+
+static int _request_matcher_forward_proxy(gcoap_listener_t *listener,
+                                          const coap_resource_t **resource,
+                                          coap_pkt_t *pdu);
+static ssize_t _forward_proxy_handler(coap_pkt_t* pdu, uint8_t *buf,
+                                      size_t len, void *ctx);
+
+const coap_resource_t forward_proxy_resources[] = {
+    { "/", COAP_IGNORE, _forward_proxy_handler, NULL },
+};
+
+gcoap_listener_t forward_proxy_listener = {
+    &forward_proxy_resources[0],
+    ARRAY_SIZE(forward_proxy_resources),
+    GCOAP_SOCKET_TYPE_UNDEF,
```

Comment:
As the proxy currently does not support CoAPS, this should be
```suggestion
    GCOAP_SOCKET_TYPE_UDP,
```